instrumentation/graphql-java-12.0/testing/src/main/java/io/opentelemetry/instrumentation/graphql/AbstractGraphqlTest.java

```diff
@@ -0,0 +1,304 @@
+            trace ->
+                trace.hasSpansSatisfyingExactly(
+                    span ->
+                        span.hasName("QUERY")
+                            .hasKind(SpanKind.INTERNAL)
+                            .hasNoParent()
+                            .hasAttributesSatisfying(
+                                attrs -> {
+                                  assertThat(attrs)
+                                      .containsEntry("graphql.operation.type", "QUERY");
+                                  String querySource =
+                                      String.valueOf(
+                                          attrs
+                                              .asMap()
+                                              .get(AttributeKey.stringKey("graphql.source")));
+                                  String normalized = querySource.replaceAll("(?s)\\s+", " ");
+                                  assertThat(normalized)
+                                      .isEqualTo("query { bookById(id: ?) { name } }");
+                                })));
+  }
+
+  @Test
+  void parseError() {
+    ExecutionResult result = graphql.execute("query foo bar");
+
+    assertThat(result.getErrors()).hasSize(1);
+    assertThat(result.getErrors().get(0).getErrorType().toString()).isEqualTo("InvalidSyntax");
+
+    getTesting()
+        .waitAndAssertTraces(
+            trace ->
+                trace.hasSpansSatisfyingExactly(
+                    span ->
+                        span.hasName("GraphQL Query")
+                            .hasKind(SpanKind.INTERNAL)
+                            .hasNoParent()
+                            .hasAttributesSatisfying(Attributes::isEmpty)
+                            .hasEventsSatisfyingExactly(
+                                event ->
+                                    event
+                                        .hasName("exception")
+                                        .hasAttributesSatisfying(
+                                            attrs -> {
+                                              assertThat(attrs)
+                                                  .containsEntry("exception.type", "InvalidSyntax");
+                                              String message =
+                                                  String.valueOf(
+                                                      attrs
+                                                          .asMap()
+                                                          .get(
+                                                              AttributeKey.stringKey(
+                                                                  "exception.message")));
+                                              assertThat(message).startsWith("Invalid Syntax");
+                                            }))));
+  }
+
+  @Test
+  void validationError() {
+    ExecutionResult result =
+        graphql.execute(
+            "" //
```

Comment:
`//` prevents spotless from moving these to a single line. Alternatively could use `// spotless:off` which would need to be enabled with `toggleOffOn()`

Reply:
Out of these two I think I'd probably prefer using `// spotless:off` for such use cases (though this one works too, I don't have a very strong opinion on that)